### PR TITLE
qt: Add diagonal directional input presets for circlepad and c-stick

### LINKED ISSUE #388: Diagonal directional buttons
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

Instead of pushing two keys on keyboard for diagonal movement just push one key

### Why would this feature be useful?

So physically disabled gamers like me who use on-screen keyboards just have to press one key for diagonal movement like on dolphin emulator

### PR DESCRIPTION
**Pull request by ghost**
Closes https://github.com/Lime3DS/Lime3DS/issues/388 

This adds new buttons to configure_input for diagonal direction presets  (i.e. upright , downleft and so on) for both the circlepad and c-stick 

Preferably should be used for single state keys but also works with game pad analog sticks

Screenshot

### CODE REVIEW

**Review by BlurrySquire — APPROVED** [review]
Works fine in Luigi's Mansion 2 and Super Mario 3D Land. Unfortunately I don't have a game that uses the c-stick so I was unable to test the c-stick. Considering that the implementation is the same it should work correctly.

**Comment by ghost** [comment]
> Works fine in Luigi's Mansion 2 and Super Mario 3D Land. Unfortunately I don't have a game that uses the c-stick so I was unable to test the c-stick. Considering that the implementation is the same it should work correctly.

You can test with a 3ds input tester homebrew 
That's what i did, and it worked on my end

**Comment by BlurrySquire** [comment]
> You can test with a 3ds input tester homebrew That's what i did, and it worked on my end

Forgot that those existed, but it is confirmed to work on more than one machine which is good.

**Comment by ghost** [comment]
Removed *some* useless code and rebased, IDK why license CI is failing on config.h tho

File: src/lime_qt/configuration/config.cpp [hunk 1]
```diff
@@ -32,20 +32,28 @@ const std::array<int, Settings::NativeButton::NumButtons> Config::default_button
...
+        Qt::Key_I, // Up
+        Qt::Key_K, // Down
+        Qt::Key_J, // Left
+        Qt::Key_L, // Right
+        Qt::Key_U, // Up-Right
+        Qt::Key_H, // Up-Left
+        Qt::Key_O, // Down-Right
+        Qt::Key_B, // Down-Left
+        Qt::Key_D  // Modifier
     },
```
Comment (OpenSauce04) [inline src/lime_qt/configuration/config.cpp]: These diagonal inputs shouldn't be (and to my knowledge aren't) bound by default, so what are these values for?

**Review by OpenSauce04 — CHANGES_REQUESTED** [review]

File: src/lime_qt/configuration/config.cpp (hunk 1 above)
Comment (ghost) [reply src/lime_qt/configuration/config.cpp]: Seems there was a misunderstanding on my end of what the aeeay actually is used for

I removed the entries and it seems to be working fine

**Review by OpenSauce04 — APPROVED** [review]
All looks fine to me. Functionality works as expected.

**Pull request merged**